processor/tailsamplingprocessor/internal/sampling/latency.go

```diff
@@ -37,10 +40,18 @@ func (l *latency) Evaluate(_ context.Context, _ pcommon.TraceID, traceData *samp
 
 	batches := traceData.ReceivedBatches
 
+	return hasSpanWithCondition(batches, l.condition()), nil
+}
+
+func (l *latency) EarlyEvaluate(_ context.Context, _ pcommon.TraceID, batch ptrace.ResourceSpans, _ *samplingpolicy.TraceData) (samplingpolicy.Decision, error) {
```

Comment:
I think this only works correctly if the upper bound is unset, otherwise we can't know whether the trace will get longer later